### ISSUE TITLE
fix: Screen reader does not announce selected item in some cases in `TreeView`

### ISSUE THREAD

**Issue by KingOfTac** [microsoft/fast#5227]
# 🐛 Bug Report

First off, I'm not entirely sure if this is an issue with `fast-tree-view` or NVDA.

Using the keyboard to navigate into a `fast-tree-view` nested item, if the nested item's parent was not the previously selected item then when the nested item is selected, the screen reader will not announce the selected event.

## 💻 Repro or Code Sample

Create a `fast-tree-view` with a nested structure.
Turn on screen reader.
Tab into `fast-tree-view`.
Select a root level tree item. The screen reader should announce that it was selected.
Using the keyboard, expand a different tree item and navigate to a nested item.
Select the nested item. The screen reader does not announce the item was selected.

Now if the nested item was selected after it's parent gets selected then the screen reader announces it properly.

## 🤔 Expected Behavior

The screen reader should announce if a nested tree item is selected regardless of where it is in the tree view in relation to the previous selected item.

## 🌍 Your Environment

* Windows 11 Build 22458.rs_prerelease.210910-1430
* Microsoft Edge Canary
* NVDA Latest

**Comment by chrisdholt** [comment]
This *sounds* like an NVDA problem. With that said, we can investigate - but the status of the selected item shouldn't depend on it's parent having *been* selected. I think this would be problematic for a number of reasons...specifically considering that it doesn't make any logical sense why the two would be related besides general hierarchy :)

Thanks for the issue, we can definitely investigate!

**Comment by chrisdholt** [comment]
> It does look an nvda issue though. After stepping through the tree view code and watching the dom updates as keyboard navigation happens, I didn't see anything that looked off between selecting an item and then selecting an immediate child item vs selecting an item and then selecting a child of a different root level item.

Yes, we've tested this across other AT as well and haven't seen the bug crop up aside from NVDA. Unfortunately, I'm not sure what we can do to mitigate this as we're providing an accurate UIA representation for the browser to translate to NVDA. I'm not aware of a good way to do target changes to specific AT to mitigate issues like this :(

**Comment by KingOfTac** [comment]
I think this issue can be closed then since there isn't anything that can be done atm. Will just need to keep an eye on nvda to see if it gets resolved in the future.

**Comment by chrisdholt** [comment]
Sounds good - thanks for digging in with us @KingofTac. Closing as this is on the NVDA side. As an FYI, I've also tagged #4127 as NVDA specific now that we have the tag
